packages/eslint-plugin/src/rules/pascal-case-enums.ts

```diff
@@ -0,0 +1,75 @@
+import { pascalCase } from "change-case";
+import { Rule } from "eslint";
+
+export default {
+    meta: {
+        type: "layout",
+        docs: {
+            description: "Enforce PascalCase for enum names",
+        },
+        fixable: "code",
+        messages: {
+            pascalCaseEnumDeclarationMessage: "Enum name should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+            pascalCaseEnumMemberMessage: "Enum member name should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+            pascalCaseEnumMemberLiteralMessage: "Enum member literal should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+        },
+    },
+    create(context) {
+        return {
+            "TSEnumDeclaration > Identifier": (identifier) => {
```

Comment:
Sadly, I couldn't find a way to type `identifier`. It seems like only https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/ast-spec exports the types. But it's an internal package that isn't publicly published.